Compiler/preproc/preprocessor.h

```diff
@@ -49,6 +49,15 @@ namespace Preprocessor {
         int _lineNumber;
         String _scriptName;
         Version _applicationVersion;
+        // A 'static' buffer for the script name when an error is
+        // reported with 'cc_error()'. We can't use the non-'static'
+        // field '_scriptName' for this purpose because the
+        // 'cc_error()' mechanism will need the buffer after this
+        // 'Preprocessor' object has been destroyed.
+        // TODO: Do away with '_scriptOfError' when 'Preprocessor'
+        // is refactored so that it doesn't use 'cc_error()' for
+        // error reporting
+        static String _scriptOfError;
```

Comment:
I forgot to mention previously, but I pushed a fix to the ags4 branch, and you do not need a static buffer for the script name anymore: 
https://github.com/adventuregamestudio/ags/commit/3facb6ad9813d81ae14864777f955448d81c23dc

This may be removed if you rebase on top of the ags4 branch.